lola_adeyemi_final_exam.py

```diff
@@ -0,0 +1,198 @@
+"""
+TASK 1
+
+Design a parent class called CFGStudent.
+
+It should have general attributes like name, surname, age, email, student id
+and methods to fetch student’s full name and student’s id.
+
+Design a child class called NanoStudent, which  inherits from CFGStudent class.
+This class should have exactly the same attributes as its parent class,
+as well as an additional one called ‘specialization’ and course grades.
+
+The child class ‘generate_id’ method should override its parent method to add the suffix ‘NANO’
+in front of the id.
+
+New methods ‘add_new_grade’ and ‘get_course_grades’ should be added.
+You can use  it as a skeleton code for your classes OR adjust it and create your own.
+
+SEE NOTES BELOW
+
+"""
+
+from random import randint
+
+
+class CFGStudent:
+
+    def __init__(self, name, surname, age, email, student_id=None):
+        self.name = name
+        self.surname = surname
+        self.age = age
+        self.email = email
+        self.student_id = self.get_id()
+
+    @staticmethod
+    def generate_id():
+        random_id = randint(1000, 10000) #create new student id using random
+        return random_id
+
+    def get_id(self):
+        return self.generate_id()
+
+    def get_fullname(self):
+        create_fullname = (self.name.title()+" "+ self.surname.title())
+        return print(create_fullname)
+
+
+class NanoStudent(CFGStudent):
+
+    def __init__(self, name, surname, email, age, specialization, course_grades=None):
+        super().__init__(name, surname, email, age,)
+        self.specialization = specialization
+        self.course_grades = {course_grades}
+
+    @staticmethod
+    def generate_id():
+        random_id = randint(1000, 10000) #create new student id using random
+        id = print("NANO{}".format(random_id))
+        return id
+        #"Example 'NANO1234' "
+
+    def add_new_grade(self, subject, grade):
+        self.course_grades.add(subject=grade)
+
+    def get_course_grades(self):
+        return self.course_grades
+
+
+
+
+############################################
+#
+# # # Example run
+#
+s = CFGStudent('Anna', 'Smith', 18, 'anna@mail.com')  # do not pass ID, it should be generated automatically
+print(s.get_fullname())
+# returns 'Anna Smith'
+print(s.student_id)
+# returns '3868' or some other random number
+
+cfg_s = NanoStudent('Software', 'Mia', 'Papandopulu', 20, 'mia@mail.com')
+print(cfg_s.get_fullname())
+# returns 'Mia Papandopulu'
+print(cfg_s.get_id())
+# returns 'NANO6180' or some other random number
+
+cfg_s.add_new_grade('theory', 95)
+cfg_s.add_new_grade('project', 78)
+print(cfg_s.get_course_grades())
+#returns {'theory': 95, 'project': 78}
+
```

Comment:
You are on the right path here, but seem to have some errors throughout. Especially the add method.